core/src/main/java/org/elasticsearch/index/query/functionscore/exp/ExponentialDecayFunctionParser.java

```diff
@@ -33,7 +34,12 @@
     }
 
     @Override
-    public ExponentialDecayFunctionBuilder getBuilderPrototype() {
-        return PROTOTYPE;
+    protected ExponentialDecayFunctionBuilder createFromBytes(String fieldName, BytesReference functionBytes) {
+        return new ExponentialDecayFunctionBuilder(fieldName, functionBytes);
+    }
+
+    @Override
+    public ExponentialDecayFunctionBuilder read(StreamInput t) throws IOException {
```

Comment:
The parser is just the thing that is registered. I could make it so you registered the parser and the reader? The the parser doesn't need to implement this read method at all and instead you just use a method reference to the constructor when you register. That sounds a lot better, actually.